File: doc-examples/mongo-reactive-example-kotlin/src/main/kotlin/example/Person.kt
package example

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import org.bson.types.ObjectId

@MappedEntity
class Person {
    @field:Id
    @GeneratedValue
    var id: ObjectId? = null
    var name: String? = null
    var age = 0

    constructor(name: String?, age: Int) : this(null, name, age) {}
    constructor(id: ObjectId?, name: String?, age: Int) {
        this.id = id
        this.name = name
        this.age = age
    }
}

File: doc-examples/mongo-example-python/src/main/python/example/Person.py
from dataclasses import dataclass, field
from typing import Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity
from org.bson.types import ObjectId


@MappedEntity
@dataclass
class Person:
    name: str
    age: int
    interests: list[str] = field(default_factory=list)
    id: Annotated[ObjectId | None, Id, GeneratedValue] = None
